server/SocketManager/socketmanager.js

```diff
@@ -1,11 +1,14 @@
 let sockets = {};
+let socket_user_map = {};
 
-const updateSocket = (usertoken, clientsocket) => {
+const updateSocket = (usertoken, clientsocket, user_id) => {
   sockets[usertoken] = clientsocket;
+  socket_user_map[usertoken] = user_id;
 };
 
 const removeSocket = usertoken => {
   delete sockets[usertoken];
+  delete socket_user_map[usertoken];
```

Comment:
In the `removeSocket` function, it would be beneficial to check if the `usertoken` exists in `socket_user_map` before attempting to delete it. This can prevent potential errors if the `usertoken` is not found.
*This comment was auto-generated by **Code Bot** to provide suggestions and improvements.*